source/NVDAObjects/UIA/edge.py

```diff
@@ -213,31 +269,29 @@ def _getTextWithFieldsForUIARange(self,rootElement,textRange,formatConfig,includ
 		for parentElement,field in parents:
 			if log.isEnabledFor(log.DEBUG):
 				log.debug("parentElement: %s"%parentElement.currentLocalizedControlType)
-			log.debug("is enclosingElement: False")
 			tempRange.MoveEndpointByRange(UIAHandler.TextPatternRangeEndpoint_Start,tempRange,UIAHandler.TextPatternRangeEndpoint_End)
-			try:
-				parentRange=self.obj.UIATextPattern.rangeFromChild(parentElement)
-			except COMError:
-				log.debug("Error fetching parent range")
-				parentRange=None
+			parentRange=self.obj.getNormalizedUIATextRangeFromElement(parentElement)
 			if parentRange:
-				self._normalizeUIARange(parentRange)
 				clippedStart=parentRange.CompareEndpoints(UIAHandler.TextPatternRangeEndpoint_Start,textRange,UIAHandler.TextPatternRangeEndpoint_Start)<0
 				tempRange.MoveEndpointByRange(UIAHandler.TextPatternRangeEndpoint_End,parentRange,UIAHandler.TextPatternRangeEndpoint_End)
 				if tempRange.CompareEndpoints(UIAHandler.TextPatternRangeEndpoint_End,textRange,UIAHandler.TextPatternRangeEndpoint_End)>0:
 					tempRange.MoveEndpointByRange(UIAHandler.TextPatternRangeEndpoint_End,textRange,UIAHandler.TextPatternRangeEndpoint_End)
 					clippedEnd=True
 				else:
 					clippedEnd=False

```

Comment:
Are these clippedStart and clippedEnd variables only needed if field is not None? if so, we could make setting them conditional.